helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt

```diff
@@ -38,22 +39,26 @@ internal object PathExcludeGenerator {
         val files = filePaths.mapTo(mutableSetOf()) { File(it) }
         val dirs = getAllDirs(files)
 
-        val pathExcludes = mutableSetOf<PathExclude>()
+        val dirsToExclude = mutableMapOf<File, PathExcludeReason>()
 
         dirs.forEach { dir ->
             PATH_EXCLUDES_REASON_FOR_DIR_NAME[dir.name]?.let { reason ->
-                pathExcludes += PathExclude(
-                    pattern = "${dir.path}/**",
-                    reason = reason
-                )
+                dirsToExclude += dir to reason
             }
         }
 
-        val filesForPathExcludes = pathExcludes.associateWith { pathExcludeExclude ->
-            files.filter { pathExcludeExclude.matches(it.path) }.toSet()
+        val result = mutableSetOf<PathExclude>()
+
+        dirsToExclude.forEach { (dir, reason) ->
+            if (dir.getAncestorFiles().intersect(dirsToExclude.keys).isEmpty()) {
+                result += PathExclude(
+                      pattern = "${dir.path}/**",
+                      reason = reason
+                )
+            }
         }
 
-        return greedySetCover(filesForPathExcludes).toList()
+        return result.toList()
```

Comment:
Is `greedySetCover` now still used, or can it be removed?

Reply:
This is still used by another helper-cli command. So, it cannot be removed.
Note: I'll likely need to use it again for the path exclude generation.